Fix net-server.c when AI_NUMERICSERV is undefined

Some macOS versions do not define AI_NUMERICSERV.  Other source files
check whether it is defined before using it; do so here as well.

https://github.com/krb5/krb5/commit/f8ecc0ae74c7ebd84f042e28079aa6b4b2ae405c
--- lib/apputils/net-server.c.orig	2022-05-25 20:35:20.000000000 -0500
+++ lib/apputils/net-server.c	2022-06-02 18:01:24.000000000 -0500
@@ -828,7 +828,10 @@
      * resolution. */
     memset(&hints, 0, sizeof(struct addrinfo));
     hints.ai_family = AF_UNSPEC;
-    hints.ai_flags = AI_PASSIVE | AI_NUMERICSERV;
+    hints.ai_flags = AI_PASSIVE;
+#ifdef AI_NUMERICSERV
+    hints.ai_flags |= AI_NUMERICSERV;
+#endif
 
     /* Add all the requested addresses. */
     for (i = 0; i < bind_addresses.n; i++) {
